### PR TITLE
Reset all Uri fields when reusing an instance

### LINKED ISSUE #41177: URI creation for certain URIs regressed from Preview 4 to Preview 5
### Description

Some pack URIs that work correctly in Preview 4, throw in Preview 5.  Replacing the most recent RC1 SDK System.Private.Uri.dll with the version from Preview 4 resolves the issue.  This is causing some failures in existing WPF tests and applications.  

```cs
using System;
using System.IO.Packaging;

namespace packagingrepro
{
    class Program
    {
        static void Main(string[] args)
        {
            // Exception thrown: 'System.UriFormatException' in System.Private.Uri.dll
            Uri[] uriArray = new Uri[2];
            uriArray[0] = CreatePackUri("http://www.nop.com/files*/photos.package", "/my pics/funЩ.jpg");
            uriArray[1] = CreatePackUri("http://www.qrs.com/fi%20l&es/photos.package", "/my%20pics/funв.jpg");
        }

        private static Uri CreatePackUri(string uri, string partUri)
        {
            return PackUriHelper.Create(new Uri(uri), PackUriHelper.CreatePartUri(new Uri(partUri, UriKind.Relative)), null);
        }
    }
}
```

### Configuration

The failure repros on Windows 7, Windows Blue, Windows Server, and Windows 10 on x86 and x64.

### Regression?

Preview 4 (5.0.0-preview.4.20251.6) works as expected.  The failure repros in Preview 5 onward. (3.1 is unaffected.) 

### Other information

> System.UriFormatException: 'Invalid URI: The Authority/Host could not be parsed.'

The error is set in System.Uri.PrivateParseMinimal. 

The exception is thrown here:

>	System.Private.Uri.dll!System.Uri.CreateThis(string uri, bool dontEscape, System.UriKind uriKind) Line 36	C#
 	System.Private.Uri.dll!System.Uri.Uri(System.Uri baseUri, System.Uri relativeUri) Line 555	C#
 	System.IO.Packaging.dll!System.IO.Packaging.PackUriHelper.Create(System.Uri packageUri, System.Uri partUri, string fragment) Line 110	C#
 	packagingrepro.dll!packagingrepro.Program.CreatePackUri(string uri, string partUri) Line 18	C#
 	packagingrepro.dll!packagingrepro.Program.Main(string[] args) Line 12	C#

### Minimal Repro

[PackagingFailureMinimal.zip](https://github.com/dotnet/runtime/files/5110666/PackagingFailureMinimal.zip)

### PR DESCRIPTION
**Pull request by MihaZupan**
Simplifying `OriginalString` to `_originalUnicodeString ?? _string` in #33044 introduced a bug affecting the `new Uri(baseUri, relativeUri)` combine operations.

The bug can be hit if:
1. The `relativeUri` contains non-ascii characters
1. The `baseUri` uses a custom (not built-in scheme) - in this case `pack`
1. A custom `UriParser` is registered for that scheme
1. That parser is a "Simple parser" - a `GenericUriParser`, and it is set up without the `GenericUriParserOptions.IriParsing` flag

The cause is that when combining the Uris, the Uri instance gets reused, but not all internal fields are reset. As `_originalUnicodeString` was not being reset, it changed the output of `OriginalString` when reused. This led to the relative Uri being used in the parsing logic instead of the constructed combined Uri.

This does not affect built-in schemes / custom parsers with the `IriParsing` flag, as `_originalUnicodeString` will be properly set during parsing (the old value won't sneak through).

Fixes #41177

### CODE REVIEW

**Comment by MihaZupan** [comment]
cc: @ryalanms

File: src/libraries/System.Private.Uri/tests/FunctionalTests/UriTests.cs [hunk 1]
```diff
@@ -709,6 +709,23 @@ public static void Uri_EmptyPort_UsesDefaultPort()
...
+            UriParser.Register(new GenericUriParser(GenericUriParserOptions.GenericAuthority), "combine-scheme", -1);
+
+            const string RelativeUriString = "/relative/uri/with/non/ascii/\u00FC";
+
+            var absoluteUri = new Uri("combine-scheme://foo", UriKind.Absolute);
+            var relativeUri = new Uri(RelativeUriString, UriKind.Relative);
+
+            new Uri(absoluteUri, relativeUri);
+            new Uri(absoluteUri, RelativeUriString);
+        }
```
Comment (karelz) [inline src/libraries/System.Private.Uri/tests/FunctionalTests/UriTests.cs:727]: Would it make sense to test the result Uri against the original? (e.g. ToStrings)

**Review by karelz — APPROVED** [review]

File: src/libraries/System.Private.Uri/tests/FunctionalTests/UriTests.cs (hunk 1 above)
Comment (MihaZupan) [reply src/libraries/System.Private.Uri/tests/FunctionalTests/UriTests.cs:727]: I can expand the test to do that, in its current form it's testing that the ctor won't throw for these inputs as it's the simplest way to validate the change.

**Review by karelz — APPROVED** [review]

**Comment by ryalanms** [comment]
Thank you, @MihaZupan and @karelz, for the fast response and fix!

**Comment by ryalanms** [comment]
cc @dotnet/wpf-developers

**Review by ManickaP — APPROVED** [review]
LGTM

**Pull request merged**